theon/src/main.rs

```diff
@@ -13,21 +13,175 @@
 #![cfg_attr(not(test), no_main)]
 #![cfg_attr(not(test), no_std)]
 
+//!
+//! Theon has two primary responsibilities:
```

Comment:
Thanks for adding this!  I was gonna take a crack at adding some comments myself as I was trying to figure out how all this works.

Reply:
No problem! I need to do another sweep and add more extensive comments.